src/lib/ui/Snackbar/Snackbar.tsx

```diff
@@ -29,29 +30,41 @@ export function Snackbar({
   autoDismiss = false,
   dismissAfterMs = 5000,
   disableCloseButton = false,
+  automaticVisibility = false,
   onClose = () => {},
 }: SnackbarProps): JSX.Element {
+  const [hidden, setHidden] = useState(visible)
+
+  useEffect(() => {
+    setHidden(!visible)
+  }, [visible])
+
   const { label: actionLabel, onClick: handleActionClick } = action ?? {}
 
+  const handleClose = useCallback(() => {
+    setHidden(true)
+    onClose()
+  }, [onClose])
```

Comment:
Surprised there isn't a linter error here
```suggestion
  }, [onClose, setHidden])
```